front/lib/resources/trigger_resource.ts

```diff
@@ -0,0 +1,184 @@
+import type { Result } from "@dust-tt/client";
+import { assertNever, Err, Ok } from "@dust-tt/client";
+import type {
+  Attributes,
+  CreationAttributes,
+  InferAttributes,
+  ModelStatic,
+  Transaction,
+} from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import { TriggerModel } from "@app/lib/models/assistant/triggers";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import type { ResourceFindOptions } from "@app/lib/resources/types";
+import {
+  createOrUpdateAgentScheduleWorkflow,
+  deleteAgentScheduleWorkflow,
+} from "@app/temporal/agent_schedule/client";
+import { normalizeError } from "@app/types";
+import type { TriggerType } from "@app/types/assistant/triggers";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unsafe-declaration-merging
+export interface TriggerResource extends ReadonlyAttributesType<TriggerModel> {}
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class TriggerResource extends BaseResource<TriggerModel> {
+  static model: ModelStatic<TriggerModel> = TriggerModel;
+
+  constructor(
+    model: ModelStatic<TriggerModel>,
+    bloc: Attributes<TriggerModel>
+  ) {
+    super(TriggerModel, bloc);
```

Comment:
nit: `bloc` -> `blob`
```suggestion
    blob: Attributes<TriggerModel>
  ) {
    super(TriggerModel, blob);
```